src/librustdoc/core.rs

```diff
@@ -326,14 +346,8 @@ pub(crate) fn run_global_ctxt(
     // typeck function bodies or run the default rustc lints.
     // (see `override_queries` in the `config`)
 
-    // HACK(jynelson) this calls an _extremely_ limited subset of `typeck`
+    // HACK(jynelson) this calls an _extremely_ limited subset of what the `analysis` query does
     // and might break if queries change their assumptions in the future.
-
-    // NOTE: This is copy/pasted from typeck/lib.rs and should be kept in sync with those changes.
-    tcx.sess.time("item_types_checking", || {
-        tcx.hir().for_each_module(|module| tcx.ensure().check_mod_item_types(module))
-    });
```

Comment:
The performance improvement is almost entirely due to this change. We stop checking item signatures.